### ISSUE TITLE
[UX] Permissions page: Make permission groups rows collapsible/expandable.

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#1046]
This is a request to actually bring [Better Permissions](https://www.drupal.org/project/better_perms) module functionality in core:

![better_permissions_d6](https://cloud.githubusercontent.com/assets/2423362/8466785/da82971a-2099-11e5-801c-5e290840c253.png)

The module is fairly simple, but was never ported to D7/D8 (https://www.drupal.org/node/937208), so the UI in the screenshot might not impress at first.

I think the addition of a global expand/collapse link somewhere at the top + making each permission group expandable/collapsible would be great. I'm not sure about the way Better Permissions expands/collapses rows though... 

- when all rows are collapsed, selecting a row expands it
- but when selecting another row, the previously expanded one collapses
- so only one expanded row at a time

**Comment by quicksketch** [comment]
I'm not sure this is necessary in combination with filtering: https://github.com/backdrop/backdrop-issues/issues/463

Assuming that searching for "Taxonomy" would show all permissions within the Taxonomy module category, even if they didn't include the word Taxonomy.

**Comment by klonos** [comment]
Yeah, this feature is more in order to allow for example viewing more than one "category" at a time while hiding the rest. So in your example how would one for instance limit the view to taxonomy module related permissions + say node module permissions? Would in that case typing "taxonomy , node" work? Even if so, is that (typing filter terms) more productive/user-friendly than collapsing all and then expanding only taxonomy and node permissions?
